pkg/utils/utils_test.go

```diff
@@ -143,3 +149,43 @@ func TestTrimSpaceAndToLower_ReturnsExpected(t *testing.T) {
 		}
 	}
 }
+
```

Comment:
Are you planning adding an E2E as well?

Reply:
this pr doesn't introduce actual namespace support, e2e test will not cover the namespaced path. I plan to add e2e tests once we fully support multi-tenancy model.